### PR TITLE
Add ability to refer to aliases of endpoints

### PR DESCRIPTION
**Pull request by illicitonion**

### CODE REVIEW

File: templating/matrix_templates/templates/http-api.tmpl [hunk 1]
```diff
@@ -1,5 +1,10 @@
 ``{{endpoint.method}} {{endpoint.path}}``
 {{(5 + (endpoint.path | length) + (endpoint.method | length)) * title_kind}}
+{% if "alias_for_path" in endpoint -%}
+``{{endpoint.path}}`` is an alias for `{{endpoint.alias_for_path}}`_.
```
Comment (Kegsay) [inline templating/matrix_templates/templates/http-api.tmpl]: Nitpick: Can we use an alternative for "alias" (which is a loaded term already for room aliases)?

**Comment by Kegsay** [comment]
This looks good, but I'm wondering if we should expand this slightly in scope to be "links" as in `See also: /foo/bar`, perhaps with a key for detailing the relation between the two endpoints which can be used as a "hint" for saying "this is an alias for" vs "see also". E.g.
```javascript
x-links:
  -  link: "post-matrix-client-api-v1-rooms-roomid-join"
     path: /join/{roomId}
     reason: "alias"
```

This would allow us to add links for reasons other than aliases (related APIs, perhaps another API which we mentioned in the summary of this API, etc) - It's a bit feature creepy I'll give you that, but I reckon it will happen sooner or later anyway as we want to link between different APIs.

**Comment by illicitonion** [comment]
I'm not convinced there's a general way we'd want to display links other than aliases, because we'd probably want the lines inline with text describing them, at which point we might as well just use inline links.

I'd say let's leave this as special-cased for aliases for now, and the first time we want to generalise this, we should do it when we have a concrete use case.

**Comment by Kegsay** [comment]
True, we do have inline links available to us. Okay, will wait and see if this comes up again.

**Pull request merged**